### PR TITLE
Add ability to skip large files

### LINKED ISSUE #424: [Feature request] Add ability to skip large files when creating Repository
When instantiating a `Repository` object, it would be great to skip large files, similar to how you can prepend a git clone with the following env var: GIT_LFS_SKIP_SMUDGE=1. This is handy when updating existing checkpoints from the hub (otherwise it first takes a long time to instantiate the `Repository` object).

### PR DESCRIPTION
**Pull request by NielsRogge**
Fixes #424 

Started a draft to add the ability to skip large files when git cloning a repo from the hub (as I need it ASAP).

To do:

- [ ] add corresponding test

### CODE REVIEW

File: src/huggingface_hub/repository.py [hunk 1]
```diff
@@ -365,6 +365,7 @@ def __init__(
         git_email: Optional[str] = None,
         revision: Optional[str] = None,
         private: bool = False,
+        skip_lfs: bool = False,
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:368]: The `skip_lfs` argument isn't super clear, what about `download_lfs_fles` set to `True` by default?

File: src/huggingface_hub/repository.py [hunk 2]
```diff
@@ -384,7 +385,7 @@ def __init__(
                 repository url (e.g. ``'https://huggingface.co/philschmid/playground-tests'``).
             repo_type (``str``, `optional`):
                 To set when creating a repo: et to "dataset" or "space" if creating a dataset or space, default is model.
-            use_auth_token (``str`` or ``bool``, `optional`, defaults ``None``):
+            use_auth_token (``str`` or ``bool``, `optional`, defaults to ``True``):
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:388]: Good, thanks

File: src/huggingface_hub/repository.py [hunk 3]
```diff
@@ -566,9 +570,12 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
             # checks if repository is initialized in a empty repository or in one with files
             if len(os.listdir(self.local_dir)) == 0:
                 logger.warning(f"Cloning {clean_repo_url} into local empty directory.")
+                prefix = ""
+                if self.skip_lfs:
+                    prefix = "GIT_LFS_SKIP_SMUDGE=1 "
```
Comment (LysandreJik) [inline src/huggingface_hub/repository.py:575]: The `subprocess.run` method takes as argument the `env` parameter, which contains the values for the environment variables. I'd rather we leverage this as it's the most cross-platform way to handle things. 

See the docs here https://docs.python.org/3/library/subprocess.html

Example usage here https://stackoverflow.com/questions/2231227/python-subprocess-popen-with-a-modified-environment

**Review by LysandreJik — COMMENTED** [review]
Great, thanks for your PR! Left a few comments.

Please also write tests leveraging the `staging` endpoint, like it's already done for other `Repository` tests. In order to test this, please use tiny files that are sill LFS tracked (thanks to their extension, for example). You can check for the size of the file to check if the pointer was downloaded or if the full file was downloaded.

File: src/huggingface_hub/repository.py (hunk 1 above)
Comment (NielsRogge) [reply src/huggingface_hub/repository.py]: I've renamed it to `skip_lfs_files`, if that's ok for you

Comment (julien-c) [reply src/huggingface_hub/repository.py]: Your call, but I also personally think that `skip_lfs` (or `skip_lfs_files`) defaulting to False is clearer (and consistent w/ the env var)

Comment (LysandreJik) [reply src/huggingface_hub/repository.py]: Yes, sounds good

**Comment by LysandreJik** [comment]
Needed this for another PR, will add the tests in the corresponding PR.

**Pull request merged**